### ISSUE TITLE
pip install broken on macOS / Xcode 9.0

### ISSUE THREAD

**Issue by morungos** [ansible/ansible#31193]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

Core

##### ANSIBLE VERSION
```
2.4.0.0
```
(Probably. Install busted so it's hard to tell.)

##### CONFIGURATION

Not got that far

##### OS / ENVIRONMENT

macOS Sierra 10.12.6 (16G29)
Xcode 9.0 installed

##### SUMMARY

Can't install: `sudo pip install ansible --upgrade` crashes on a missing `opensslv.h` file.

NOTE: Possibly a duplicate of (closed) #16120, but on macOS, there isn't an openssl-devel that's around to install. 

##### STEPS TO REPRODUCE

```
$ sudo pip install ansible --upgrade
```

##### EXPECTED RESULTS

Ansible gets installed. 

##### ACTUAL RESULTS

```
reading manifest file 'src/cryptography.egg-info/SOURCES.txt'

reading manifest template 'MANIFEST.in'

no previously-included directories found matching 'docs/_build'

warning: no previously-included files matching '*' found under directory 'vectors'

writing manifest file 'src/cryptography.egg-info/SOURCES.txt'

running build_ext

generating cffi module 'build/temp.macosx-10.12-intel-2.7/_padding.c'

creating build/temp.macosx-10.12-intel-2.7

generating cffi module 'build/temp.macosx-10.12-intel-2.7/_constant_time.c'

generating cffi module 'build/temp.macosx-10.12-intel-2.7/_openssl.c'

building '_openssl' extension

creating build/temp.macosx-10.12-intel-2.7/build

creating build/temp.macosx-10.12-intel-2.7/build/temp.macosx-10.12-intel-2.7

cc -fno-strict-aliasing -fno-common -dynamic -arch x86_64 -arch i386 -g -Os -pipe -fno-common -fno-strict-aliasing -fwrapv -DENABLE_DTRACE -DMACOSX -DNDEBUG -Wall -Wstrict-prototypes -Wshorten-64-to-32 -DNDEBUG -g -fwrapv -Os -Wall -Wstrict-prototypes -DENABLE_DTRACE -arch x86_64 -arch i386 -pipe -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c build/temp.macosx-10.12-intel-2.7/_openssl.c -o build/temp.macosx-10.12-intel-2.7/build/temp.macosx-10.12-intel-2.7/_openssl.o -Wconversion -Wno-error=sign-conversion

build/temp.macosx-10.12-intel-2.7/_openssl.c:483:10: fatal error: 'openssl/opensslv.h' file not found

#include <openssl/opensslv.h>

         ^~~~~~~~~~~~~~~~~~~~

1 error generated.

error: command 'cc' failed with exit status 1
```

**Comment by sivel** [comment]
You appear to be missing the openssl headers required to build the cryptography module.

The openssl development headers can be installed via homebrew, by using `brew install openssl`.

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Comment by morungos** [comment]
It's already installed :-(

```
Morag:turalt-public-jekyll stuart$ brew install openssl
Error: openssl 1.0.2k is already installed
To upgrade to 1.0.2l, run `brew upgrade openssl`
```

Also, even if I upgrade openssl, same error.

**Comment by SpyderAlba** [comment]
me too

**Comment by SleeplessAnnoyedNerd** [comment]
Hi,

you could try...

```
export CPPFLAGS=-I/usr/local/opt/openssl/include
export LDFLAGS=-L/usr/local/opt/openssl/lib
```

At least it helped for me. (See also https://github.com/pyca/cryptography/issues/3489.)

Kind regards,
Dietmar

**Comment by heijmans** [comment]
Updating pip worked for me:

    sudo pip install --upgrade pip
    sudo pip install ansible

It seems that easy_install installs an old version of pip (6.0.x) while pip 9.0.x is available. pip 9.0.x can install ansible without problems on my system (macOS 10.13.2). I would strongly suggest to document this in (the macOS part of) the installation docs.

**Comment by morungos** [comment]
I did encounter this issue again, recently, and a total clean of homebrew and manually removing a bunch of git-installed ansibles from past days (when that was recommended) fixed the issues. My concern was more that I needed to know UNIX environment variables for C code to get ansible to install. Dependence on "someone else's python" is always going to be a risk point, and an embedded install which contained its own python for macOS would be much easier. 

I would love a more standalone installer that dropped things into sensible places and remained somewhat isolated from whatever I've managed to do with python.
